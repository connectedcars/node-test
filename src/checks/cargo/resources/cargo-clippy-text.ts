import { CargoMessage } from '../cargo-types'

export const cargoClippyFailedOutput: CargoMessage[] = [
  {
    reason: 'compiler-message',
    package_id: 'can 0.1.0 (path+file:///app/can)',
    target: {
      kind: ['lib'],
      crate_types: ['lib'],
      name: 'can',
      src_path: '/app/can/src/lib.rs',
      edition: '2018',
      doctest: true
    },
    message: {
      rendered:
        'error: this comparison involving the minimum or maximum element for this type contains a case that is always true or always false\n   --> can/src/can_session.rs:391:12\n    |\n391 |         if 100 > i32::MAX {}\n    |            ^^^^^^^^^^^^^^\n    |\n    = note: `-D clippy::absurd-extreme-comparisons` implied by `-D clippy::all`\n    = help: because `i32::MAX` is the maximum value for this type, this comparison is always false\n    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#absurd_extreme_comparisons\n\n',
      children: [
        {
          children: [],
          code: null,
          level: 'note',
          message: '`-D clippy::absurd-extreme-comparisons` implied by `-D clippy::all`',
          rendered: null,
          spans: []
        },
        {
          children: [],
          code: null,
          level: 'help',
          message: 'because `i32::MAX` is the maximum value for this type, this comparison is always false',
          rendered: null,
          spans: []
        },
        {
          children: [],
          code: null,
          level: 'help',
          message:
            'for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#absurd_extreme_comparisons',
          rendered: null,
          spans: []
        }
      ],
      code: {
        code: 'clippy::absurd_extreme_comparisons',
        explanation: null
      },
      level: 'error',
      message:
        'this comparison involving the minimum or maximum element for this type contains a case that is always true or always false',
      spans: [
        {
          byte_end: 13338,
          byte_start: 13324,
          column_end: 26,
          column_start: 12,
          expansion: null,
          file_name: 'can/src/can_session.rs',
          is_primary: true,
          label: null,
          line_end: 391,
          line_start: 391,
          suggested_replacement: null,
          suggestion_applicability: null,
          text: [
            {
              highlight_end: 26,
              highlight_start: 12,
              text: '        if 100 > i32::MAX {}'
            }
          ]
        }
      ]
    }
  },
  {
    reason: 'compiler-message',
    package_id: 'can 0.1.0 (path+file:///app/can)',
    target: {
      kind: ['lib'],
      crate_types: ['lib'],
      name: 'can',
      src_path: '/app/can/src/lib.rs',
      edition: '2018',
      doctest: true
    },
    message: {
      rendered: 'error: aborting due to previous error\n\n',
      children: [],
      code: null,
      level: 'error',
      message: 'aborting due to previous error',
      spans: []
    }
  }
]
