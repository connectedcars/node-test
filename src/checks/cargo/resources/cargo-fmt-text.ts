import { CargoFmtFile } from '../cargo-types'

export const cargoFmtFailedOutput: CargoFmtFile[] = [
  {
    name: '/checks-test/src/foo/bar.rs',
    mismatches: [
      {
        original_begin_line: 1,
        original_end_line: 7,
        expected_begin_line: 1,
        expected_end_line: 2,
        original: '\n\npub fn bar(\n\n)\n{\n    println!  (  "bar"  )  ;',
        expected: 'pub fn bar() {\n    println!("bar");'
      }
    ]
  },
  {
    name: '/checks-test/src/foo/mod.rs',
    mismatches: [
      {
        original_begin_line: 1,
        original_end_line: 5,
        expected_begin_line: 1,
        expected_end_line: 1,
        original: '\n\nmod bar\n;\n',
        expected: 'mod bar;'
      }
    ]
  },
  {
    name: '/checks-test/src/main.rs',
    mismatches: [
      {
        original_begin_line: 9,
        original_end_line: 9,
        expected_begin_line: 9,
        expected_end_line: 9,
        original: '',
        expected: 'mod foo;'
      },
      {
        original_begin_line: 10,
        original_end_line: 10,
        expected_begin_line: 11,
        expected_end_line: 12,
        original: ' mod foo  ;',
        expected: 'fn main() {\n    let foo = foo();'
      },
      {
        original_begin_line: 12,
        original_end_line: 16,
        expected_begin_line: 14,
        expected_end_line: 14,
        original: '\nfn main()\n{\n      let foo = foo();\n',
        expected: ''
      },
      {
        original_begin_line: 19,
        original_end_line: 19,
        expected_begin_line: 16,
        expected_end_line: 16,
        original: '',
        expected: ''
      },
      {
        original_begin_line: 27,
        original_end_line: 28,
        expected_begin_line: 23,
        expected_end_line: 23,
        original: '\n',
        expected: ''
      },
      {
        original_begin_line: 35,
        original_end_line: 38,
        expected_begin_line: 29,
        expected_end_line: 32,
        original: '\n\n    if { let b = false; b } {}\n',
        expected: '    if {\n        let b = false;\n        b\n    } {}'
      },
      {
        original_begin_line: 56,
        original_end_line: 56,
        expected_begin_line: 50,
        expected_end_line: 50,
        original: '    assert!(false);',
        expected: '        assert!(false);'
      },
      {
        original_begin_line: 71,
        original_end_line: 72,
        expected_begin_line: 65,
        expected_end_line: 65,
        original: '\n',
        expected: ''
      },
      {
        original_begin_line: 74,
        original_end_line: 75,
        expected_begin_line: 66,
        expected_end_line: 66,
        original: '\n',
        expected: ''
      }
    ]
  }
]
