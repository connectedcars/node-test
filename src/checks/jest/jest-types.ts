interface JestData {
  filePath: string
  messages: EslintMessage[]
  errorCount: number
  warningCount: number
  fixableErrorCount: number
  fixableWarningCount: number
  source?: string
}

interface EslintFix {
  range: number[]
  text: string
}

interface EslintMessage {
  ruleId?: string
  fatal?: boolean
  severity: number
  message: string
  line?: number
  endLine?: number
  column?: number
  endColumn?: number
  nodeType?: string | null
  source?: string
  fix?: EslintFix
}
