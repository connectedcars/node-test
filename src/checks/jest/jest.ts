import {
  CheckAnnotation,
  CheckAnnotationLevel,
  CheckConversionError,
  CheckRunCompleted,
  MAX_LINE_LENGTH,
  MAX_OUTPUT_LENGTH
} from '../checks-common'

export interface FormattedTestResults {
  numFailedTests: number
  numFailedTestSuites: number
  numPassedTests: number
  numPassedTestSuites: number
  numPendingTests: number
  numPendingTestSuites: number
  numRuntimeErrorTestSuites: number
  numTodoTests: number
  numTotalTests: number
  numTotalTestSuites: number
  openHandles: string[]
  snapshot: SnapshotSummary
  startTime: number
  success: boolean
  testResults: Array<FormattedTestResult>
  wasInterrupted: boolean
}

export interface FormattedTestResult {
  message: string
  name: string
  summary: string
  status: 'failed' | 'passed'
  startTime: number
  endTime: number
  coverage?: unknown
  assertionResults: Array<FormattedAssertionResult>
}

interface Location {
  column: number
  line: number
}

export interface FormattedAssertionResult {
  ancestorTitles: Array<string>
  duration?: number
  failureDetails?: unknown[]
  failureMessages: Array<string>
  fullName: string
  invocations?: number
  location?: Location | null
  numPassingAsserts?: number
  retryReasons?: unknown[]
  status: 'passed' | 'failed' | 'skipped' | 'pending' | 'todo' | 'disabled'
  title: string
}

export interface SnapshotSummary {
  added: number
  didUpdate: boolean
  failure: boolean
  filesAdded: number
  filesRemoved: number
  filesRemovedList: Array<string>
  filesUnmatched: number
  filesUpdated: number
  matched: number
  total: number
  unchecked: number
  uncheckedKeysByFile: Array<UncheckedSnapshot>
  unmatched: number
  updated: number
}

export interface UncheckedSnapshot {
  filePath: string
  keys: Array<string>
}

export interface JestInput {
  sha: string
  data: FormattedTestResults
  name?: string
}

interface CreateCheckAnnotationOptions {
  file: string
  message: string
  title: string
  status: string
  rawDetails: string
  location?: Location | null
}

function getRelativePath(path: string): string {
  const match = path.match(/^.*\/((?:src|bin)\/.+)$/)
  // Path needs to be set so default to a file we know exists
  return match && match.length === 2 ? match[1] : 'jest.config.js'
}

function getLocation(message: string, location?: Location | null): number {
  if (location) {
    return location.line
  }

  let start_line = 1
  const lineColumnMatch = message.match(/sx?:(\d+):(\d+)\)/)

  if (lineColumnMatch) {
    start_line = parseInt(lineColumnMatch[1])
  }

  return start_line
}

function createCheckAnnotation({
  file,
  message,
  title,
  status,
  rawDetails,
  location
}: CreateCheckAnnotationOptions): CheckAnnotation {
  const relPath = getRelativePath(file)
  const start_line = getLocation(message, location)
  let annotation_level: CheckAnnotationLevel

  switch (status) {
    case 'disabled':
      annotation_level = 'notice'
      break
    case 'failed':
      annotation_level = 'failure'
      break
    case 'pending':
      annotation_level = 'notice'
      break
    case 'skipped':
      annotation_level = 'notice'
      break
    case 'todo':
      annotation_level = 'notice'
      break
    default:
      annotation_level = 'notice'
  }

  const annotation: CheckAnnotation = {
    start_line,
    end_line: start_line,
    annotation_level,
    title,
    message: message || 'unknown issue',
    path: relPath,
    raw_details: rawDetails
  }

  // Try to keep the output small so we avoid truncation
  let len = JSON.stringify(annotation, null, 2).length

  if (len > MAX_LINE_LENGTH) {
    // Try to only remove raw details
    annotation.raw_details = `Line too large (${len} B)`
    // Check whether the output is still too large
    len = JSON.stringify(annotation, null, 2).length
    if (len > MAX_LINE_LENGTH) {
      // Remove the message as well
      annotation.message = `Line too large (${len} B)`
    }
  }

  return annotation
}

export function jestCheck({ data, sha, name = 'jest' }: JestInput): CheckRunCompleted {
  try {
    const result: CheckRunCompleted = {
      name,
      head_sha: sha,
      conclusion: 'neutral',
      status: 'completed',
      completed_at: new Date().toISOString(),
      output: {
        title: 'No tests found',
        summary: 'No tests found',
        annotations: []
      }
    }

    if (!data) {
      return result
    }

    result.conclusion = data.success && !data.snapshot.failure ? 'success' : 'failure'

    if (result.output) {
      // There are no assertion results when the error did not originate
      // from a test assertion, for example from an import cycle
      result.output.annotations = data.testResults
        .filter(testResult => testResult.assertionResults.length === 0)
        .map(testResult =>
          createCheckAnnotation({
            file: testResult.name,
            title: getRelativePath(testResult.name),
            message: testResult.message,
            status: testResult.status,
            rawDetails: JSON.stringify(testResult, null, 2)
          })
        )

      result.output.annotations.push(
        ...data.testResults
          .flatMap(results => {
            return results.assertionResults.map(assertionResult => {
              return {
                ...assertionResult,
                file: results.name
              }
            })
          })
          .filter(assertionResult => !['passed'].includes(assertionResult.status))
          .map<CheckAnnotation>(assertionResult => {
            const failureMessage = assertionResult.failureMessages?.join('\n')
            const message =
              assertionResult.status === 'pending' ? `skipped test '${assertionResult.title}'` : failureMessage
            const title = [...assertionResult.ancestorTitles, assertionResult.title].join(' - ')

            return createCheckAnnotation({
              file: assertionResult.file,
              status: assertionResult.status,
              title,
              message,
              rawDetails: JSON.stringify(assertionResult, null, 2),
              location: assertionResult.location
            })
          })
          // Limit to 50 annotations as this is the max per post for github
          .slice(0, 50)
      )

      result.output.summary = `${data.numPassedTests} of ${data.numTotalTests} tests passed`
      if (data.snapshot.failure) {
        if (data.snapshot.unchecked > 0) {
          result.output.summary += ` (with ${data.snapshot.unchecked} snapshots obsolete)`
          for (const { filePath, keys } of data.snapshot.uncheckedKeysByFile) {
            if (!result.output.annotations) {
              result.output.annotations = []
            }
            result.output.annotations.push({
              start_line: 1,
              end_line: 1,
              annotation_level: 'failure',
              title: `${keys.length} snapshots obsolete`,
              message: ` - ${keys.join('\n - ')}`,
              path: getRelativePath(filePath),
              raw_details: JSON.stringify(data.snapshot, null, 2)
            })
          }
        }
      }
      // note: check numTotalTestSuites for count?
      result.output.title = result.output.summary
      // Try to keep the output small so we avoid truncation
      let len = JSON.stringify(result, null, 2).length
      if (len > MAX_OUTPUT_LENGTH) {
        for (const annotation of result.output.annotations || []) {
          // Try to only remove raw details
          annotation.raw_details = `Output too large (${len} B)`
        }
        // Check whether the output is still too large
        len = JSON.stringify(result, null, 2).length
        if (len > MAX_OUTPUT_LENGTH) {
          if (result.output.annotations.length > 0) {
            // Distribute the truncation evenly by half
            for (const annotation of result.output.annotations) {
              const amountToTruncate = Math.floor(annotation.message.length / 2)
              annotation.message = annotation.message.slice(0, amountToTruncate)
              if (annotation.message.length === 0) {
                annotation.message = 'Output too large (unable to truncate)'
              }
            }
          }
        }

        // Check whether the output is still too large. Remove the message that was truncated
        len = JSON.stringify(result, null, 2).length
        if (len > MAX_OUTPUT_LENGTH) {
          for (const annotation of result.output.annotations) {
            annotation.message = `Output too large (${len} B)`
          }
        }
      }
    }

    return result
  } catch (e) {
    throw new CheckConversionError('jest', { data, sha }, e)
  }
}
