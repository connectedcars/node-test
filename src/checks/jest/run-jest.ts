import { FormattedTestResults } from '@jest/test-result/build/types'

import { RunProcess } from '../../unix/run-process'

export async function runJestBin(): Promise<FormattedTestResults> {
  const cmd = new RunProcess('./node_modules/.bin/jest', ['--silent', '--no-color', '--json'], {
    env: { ...process.env, TZ: 'UTC' }
  })
  const data: Buffer[] = []
  cmd.stdout?.on('data', chunk => {
    data.push(chunk)
  })
  await cmd.waitForStarted()
  await cmd.waitForExit()
  const json = Buffer.concat(data).toString('utf8')
  return JSON.parse(json)
}

export async function runJest(): Promise<FormattedTestResults> {
  const cmd = new RunProcess('jest', ['--silent', '--no-color', '--json'], {
    env: { ...process.env, TZ: 'UTC' }
  })
  const data: Buffer[] = []
  cmd.stdout?.on('data', chunk => {
    data.push(chunk)
  })
  await cmd.waitForStarted()
  await cmd.waitForExit()
  const json = Buffer.concat(data).toString('utf8')
  return JSON.parse(json)
}

export async function runReactScriptsTest(): Promise<string> {
  const cmd = new RunProcess('react-scripts test', ['--silent', '--no-color', '--json'], {
    env: { ...process.env, TZ: 'UTC', CI: 'true' }
  })
  const data: Buffer[] = []
  cmd.stdout?.on('data', chunk => {
    data.push(chunk)
  })

  await cmd.waitForStarted()
  await cmd.waitForExit()

  const json = Buffer.concat(data).toString('utf8')
  return JSON.parse(json)
}
