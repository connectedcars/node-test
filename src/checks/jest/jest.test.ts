import { FormattedTestResults, jestCheck } from './jest'
import { jestFailedOutput, jestPassedOutput, jestSkippedOutput } from './resources/jest-help-text'

describe('checks/jest', () => {
  it('processes passing jest output to checks structure', () => {
    const output = jestCheck({ data: jestPassedOutput, sha: '1234567890' })
    const expected = {
      name: 'jest',
      head_sha: '1234567890',
      conclusion: 'success',
      status: 'completed',
      completed_at: expect.stringMatching(/^\d{4}/),
      output: {
        title: '33 of 33 tests passed!',
        summary: '33 of 33 tests passed!',
        annotations: []
      }
    }
    expect(output).toStrictEqual(expected)
  })

  it('processes failing jest output to checks structure', () => {
    const output = jestCheck({ data: jestFailedOutput, sha: '1234567890' })
    expect(output).toMatchSnapshot({
      completed_at: expect.stringMatching(/^\d{4}/)
    })
  })

  it('handles empty input', () => {
    const sampleOutput = ''
    const output = jestCheck({
      data: sampleOutput as unknown as FormattedTestResults,
      sha: '1234567890'
    })
    expect(output).toStrictEqual({
      name: 'jest',
      head_sha: '1234567890',
      conclusion: 'neutral',
      status: 'completed',
      completed_at: expect.stringMatching(/^\d{4}/),
      output: {
        title: 'No tests found',
        summary: 'No tests found',
        annotations: []
      }
    })
  })

  it('should handle skipped tests', () => {
    const output = jestCheck({ data: jestSkippedOutput, sha: '1234567890' })
    expect(output).toMatchSnapshot({
      completed_at: expect.stringMatching(/^\d{4}/)
    })
  })
})
