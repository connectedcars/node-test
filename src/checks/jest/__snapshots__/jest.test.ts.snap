// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks/jest processes failing jest output to checks structure 1`] = `
Object {
  "completed_at": StringMatching /\\^\\\\d\\{4\\}/,
  "conclusion": "failure",
  "head_sha": "1234567890",
  "name": "jest",
  "output": Object {
    "annotations": Array [
      Object {
        "annotation_level": "failure",
        "end_line": 61,
        "message": "Error: expect(received).toStrictEqual(expected) // deep equality

- Expected - 1
+ Received + 1

 Object {
 \\"conclusion\\": \\"success\\",
 \\"output\\": Object {
 \\"annotations\\": Array [],
 \\"summary\\": \\"No problems found\\",
- \\"title\\": \\"No problems found BUG\\",
+ \\"title\\": \\"No problems found\\",
 },
 }
 at Object.<anonymous> (/home/jagdos/repos/node-ci-tools/src/eslint/eslint.test.ts:61:18)
 at Object.asyncJestTest (/home/jagdos/repos/node-ci-tools/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:100:37)
 at /home/jagdos/repos/node-ci-tools/node_modules/jest-jasmine2/build/queueRunner.js:45:12
 at new Promise (<anonymous>)
 at mapper (/home/jagdos/repos/node-ci-tools/node_modules/jest-jasmine2/build/queueRunner.js:28:19)
 at /home/jagdos/repos/node-ci-tools/node_modules/jest-jasmine2/build/queueRunner.js:75:41",
        "path": "src/eslint/eslint.test.ts",
        "raw_details": "{
  \\"ancestorTitles\\": [
    \\"describe/name\\"
  ],
  \\"failureMessages\\": [
    \\"Error: expect(received).toStrictEqual(expected) // deep equality\\\\n\\\\n- Expected - 1\\\\n+ Received + 1\\\\n\\\\n Object {\\\\n \\\\\\"conclusion\\\\\\": \\\\\\"success\\\\\\",\\\\n \\\\\\"output\\\\\\": Object {\\\\n \\\\\\"annotations\\\\\\": Array [],\\\\n \\\\\\"summary\\\\\\": \\\\\\"No problems found\\\\\\",\\\\n- \\\\\\"title\\\\\\": \\\\\\"No problems found BUG\\\\\\",\\\\n+ \\\\\\"title\\\\\\": \\\\\\"No problems found\\\\\\",\\\\n },\\\\n }\\\\n at Object.<anonymous> (/home/jagdos/repos/node-ci-tools/src/eslint/eslint.test.ts:61:18)\\\\n at Object.asyncJestTest (/home/jagdos/repos/node-ci-tools/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:100:37)\\\\n at /home/jagdos/repos/node-ci-tools/node_modules/jest-jasmine2/build/queueRunner.js:45:12\\\\n at new Promise (<anonymous>)\\\\n at mapper (/home/jagdos/repos/node-ci-tools/node_modules/jest-jasmine2/build/queueRunner.js:28:19)\\\\n at /home/jagdos/repos/node-ci-tools/node_modules/jest-jasmine2/build/queueRunner.js:75:41\\"
  ],
  \\"fullName\\": \\"converts successful eslint with skipped file\\",
  \\"location\\": null,
  \\"status\\": \\"failed\\",
  \\"title\\": \\"converts successful eslint with skipped file\\",
  \\"file\\": \\"/home/jagdos/repos/node-ci-tools/src/eslint/eslint.test.ts\\"
}",
        "start_line": 61,
        "title": "describe/name - converts successful eslint with skipped file",
      },
      Object {
        "annotation_level": "failure",
        "end_line": 278,
        "message": "Error: [2mexpect([22m[31mreceived[39m[2m).[22mtoEqual[2m([22m[32mexpected[39m[2m) // deep equality[22m

[32m- Expected  -  1[39m
[31m+ Received  + 17[39m

[32m- Array [][39m
[31m+ Array [[39m
[31m+   RowDataPacket {[39m
[31m+     \\"attempts\\": 0,[39m
[31m+     \\"carId\\": 124,[39m
[31m+     \\"code\\": null,[39m
[31m+     \\"createdAt\\": 2023-03-23T13:45:12.000Z,[39m
[31m+     \\"id\\": 1,[39m
[31m+     \\"primaryCanDtcId\\": 1,[39m
[31m+     \\"rawCode\\": \\"0304\\",[39m
[31m+     \\"resolution\\": null,[39m
[31m+     \\"resolved\\": 0,[39m
[31m+     \\"resolvedGeneric\\": 0,[39m
[31m+     \\"task\\": \\"{\\\\\\"ecu\\\\\\": {\\\\\\"rxId\\\\\\": \\\\\\"7E8\\\\\\", \\\\\\"txId\\\\\\": \\\\\\"7E0\\\\\\", \\\\\\"category\\\\\\": \\\\\\"01\\\\\\", \\\\\\"transport\\\\\\": \\\\\\"ISOTP\\\\\\", \\\\\\"application\\\\\\": \\\\\\"UDS\\\\\\", \\\\\\"requestsAndResponses\\\\\\": [{\\\\\\"response\\\\\\": \\\\\\"something useful\\\\\\", \\\\\\"serviceAndDid\\\\\\": \\\\\\"22F19E\\\\\\"}, {\\\\\\"response\\\\\\": \\\\\\"something useful\\\\\\", \\\\\\"serviceAndDid\\\\\\": \\\\\\"22F1A2\\\\\\"}, {\\\\\\"response\\\\\\": \\\\\\"something useful\\\\\\", \\\\\\"serviceAndDid\\\\\\": \\\\\\"22F197\\\\\\"}, {\\\\\\"response\\\\\\": \\\\\\"5902FF030400\\\\\\", \\\\\\"serviceAndDid\\\\\\": \\\\\\"1902AE\\\\\\"}]}, \\\\\\"siblings\\\\\\": [{\\\\\\"rxId\\\\\\": \\\\\\"7E8\\\\\\", \\\\\\"txId\\\\\\": \\\\\\"7E0\\\\\\", \\\\\\"category\\\\\\": \\\\\\"01\\\\\\", \\\\\\"transport\\\\\\": \\\\\\"ISOTP\\\\\\", \\\\\\"application\\\\\\": \\\\\\"OBD\\\\\\", \\\\\\"requestsAndResponses\\\\\\": [{\\\\\\"response\\\\\\": \\\\\\"49026d7976696e\\\\\\", \\\\\\"serviceAndDid\\\\\\": \\\\\\"0902\\\\\\"}]}]}\\",[39m
[31m+     \\"updatedAt\\": 2023-03-23T13:45:12.000Z,[39m
[31m+     \\"vendor\\": \\"VAG\\",[39m
[31m+   },[39m
[31m+ ][39m
    at Object.toEqual (/Users/michaelstorgaard/Sites/job-runner-reverse-engineering/src/jobs/process-vag-can-dtcs.it.test.ts:278:30)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)",
        "path": "src/eslint/eslint.test.ts",
        "raw_details": "{
  \\"ancestorTitles\\": [
    \\"jobs/process-vag-can-dtcs\\"
  ],
  \\"duration\\": 549,
  \\"failureDetails\\": [],
  \\"failureMessages\\": [
    \\"Error: \\\\u001b[2mexpect(\\\\u001b[22m\\\\u001b[31mreceived\\\\u001b[39m\\\\u001b[2m).\\\\u001b[22mtoEqual\\\\u001b[2m(\\\\u001b[22m\\\\u001b[32mexpected\\\\u001b[39m\\\\u001b[2m) // deep equality\\\\u001b[22m\\\\n\\\\n\\\\u001b[32m- Expected  -  1\\\\u001b[39m\\\\n\\\\u001b[31m+ Received  + 17\\\\u001b[39m\\\\n\\\\n\\\\u001b[32m- Array []\\\\u001b[39m\\\\n\\\\u001b[31m+ Array [\\\\u001b[39m\\\\n\\\\u001b[31m+   RowDataPacket {\\\\u001b[39m\\\\n\\\\u001b[31m+     \\\\\\"attempts\\\\\\": 0,\\\\u001b[39m\\\\n\\\\u001b[31m+     \\\\\\"carId\\\\\\": 124,\\\\u001b[39m\\\\n\\\\u001b[31m+     \\\\\\"code\\\\\\": null,\\\\u001b[39m\\\\n\\\\u001b[31m+     \\\\\\"createdAt\\\\\\": 2023-03-23T13:45:12.000Z,\\\\u001b[39m\\\\n\\\\u001b[31m+     \\\\\\"id\\\\\\": 1,\\\\u001b[39m\\\\n\\\\u001b[31m+     \\\\\\"primaryCanDtcId\\\\\\": 1,\\\\u001b[39m\\\\n\\\\u001b[31m+     \\\\\\"rawCode\\\\\\": \\\\\\"0304\\\\\\",\\\\u001b[39m\\\\n\\\\u001b[31m+     \\\\\\"resolution\\\\\\": null,\\\\u001b[39m\\\\n\\\\u001b[31m+     \\\\\\"resolved\\\\\\": 0,\\\\u001b[39m\\\\n\\\\u001b[31m+     \\\\\\"resolvedGeneric\\\\\\": 0,\\\\u001b[39m\\\\n\\\\u001b[31m+     \\\\\\"task\\\\\\": \\\\\\"{\\\\\\\\\\\\\\"ecu\\\\\\\\\\\\\\": {\\\\\\\\\\\\\\"rxId\\\\\\\\\\\\\\": \\\\\\\\\\\\\\"7E8\\\\\\\\\\\\\\", \\\\\\\\\\\\\\"txId\\\\\\\\\\\\\\": \\\\\\\\\\\\\\"7E0\\\\\\\\\\\\\\", \\\\\\\\\\\\\\"category\\\\\\\\\\\\\\": \\\\\\\\\\\\\\"01\\\\\\\\\\\\\\", \\\\\\\\\\\\\\"transport\\\\\\\\\\\\\\": \\\\\\\\\\\\\\"ISOTP\\\\\\\\\\\\\\", \\\\\\\\\\\\\\"application\\\\\\\\\\\\\\": \\\\\\\\\\\\\\"UDS\\\\\\\\\\\\\\", \\\\\\\\\\\\\\"requestsAndResponses\\\\\\\\\\\\\\": [{\\\\\\\\\\\\\\"response\\\\\\\\\\\\\\": \\\\\\\\\\\\\\"something useful\\\\\\\\\\\\\\", \\\\\\\\\\\\\\"serviceAndDid\\\\\\\\\\\\\\": \\\\\\\\\\\\\\"22F19E\\\\\\\\\\\\\\"}, {\\\\\\\\\\\\\\"response\\\\\\\\\\\\\\": \\\\\\\\\\\\\\"something useful\\\\\\\\\\\\\\", \\\\\\\\\\\\\\"serviceAndDid\\\\\\\\\\\\\\": \\\\\\\\\\\\\\"22F1A2\\\\\\\\\\\\\\"}, {\\\\\\\\\\\\\\"response\\\\\\\\\\\\\\": \\\\\\\\\\\\\\"something useful\\\\\\\\\\\\\\", \\\\\\\\\\\\\\"serviceAndDid\\\\\\\\\\\\\\": \\\\\\\\\\\\\\"22F197\\\\\\\\\\\\\\"}, {\\\\\\\\\\\\\\"response\\\\\\\\\\\\\\": \\\\\\\\\\\\\\"5902FF030400\\\\\\\\\\\\\\", \\\\\\\\\\\\\\"serviceAndDid\\\\\\\\\\\\\\": \\\\\\\\\\\\\\"1902AE\\\\\\\\\\\\\\"}]}, \\\\\\\\\\\\\\"siblings\\\\\\\\\\\\\\": [{\\\\\\\\\\\\\\"rxId\\\\\\\\\\\\\\": \\\\\\\\\\\\\\"7E8\\\\\\\\\\\\\\", \\\\\\\\\\\\\\"txId\\\\\\\\\\\\\\": \\\\\\\\\\\\\\"7E0\\\\\\\\\\\\\\", \\\\\\\\\\\\\\"category\\\\\\\\\\\\\\": \\\\\\\\\\\\\\"01\\\\\\\\\\\\\\", \\\\\\\\\\\\\\"transport\\\\\\\\\\\\\\": \\\\\\\\\\\\\\"ISOTP\\\\\\\\\\\\\\", \\\\\\\\\\\\\\"application\\\\\\\\\\\\\\": \\\\\\\\\\\\\\"OBD\\\\\\\\\\\\\\", \\\\\\\\\\\\\\"requestsAndResponses\\\\\\\\\\\\\\": [{\\\\\\\\\\\\\\"response\\\\\\\\\\\\\\": \\\\\\\\\\\\\\"49026d7976696e\\\\\\\\\\\\\\", \\\\\\\\\\\\\\"serviceAndDid\\\\\\\\\\\\\\": \\\\\\\\\\\\\\"0902\\\\\\\\\\\\\\"}]}]}\\\\\\",\\\\u001b[39m\\\\n\\\\u001b[31m+     \\\\\\"updatedAt\\\\\\": 2023-03-23T13:45:12.000Z,\\\\u001b[39m\\\\n\\\\u001b[31m+     \\\\\\"vendor\\\\\\": \\\\\\"VAG\\\\\\",\\\\u001b[39m\\\\n\\\\u001b[31m+   },\\\\u001b[39m\\\\n\\\\u001b[31m+ ]\\\\u001b[39m\\\\n    at Object.toEqual (/Users/michaelstorgaard/Sites/job-runner-reverse-engineering/src/jobs/process-vag-can-dtcs.it.test.ts:278:30)\\\\n    at runMicrotasks (<anonymous>)\\\\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\\"
  ],
  \\"fullName\\": \\"jobs/process-vag-can-dtcs does not processes TL codes\\",
  \\"invocations\\": 1,
  \\"location\\": {
    \\"line\\": 278,
    \\"column\\": 30
  },
  \\"numPassingAsserts\\": 1,
  \\"retryReasons\\": [],
  \\"status\\": \\"failed\\",
  \\"title\\": \\"does not processes TL codes\\",
  \\"file\\": \\"/home/jagdos/repos/node-ci-tools/src/eslint/eslint.test.ts\\"
}",
        "start_line": 278,
        "title": "jobs/process-vag-can-dtcs - does not processes TL codes",
      },
    ],
    "summary": "3 of 4 tests passed",
    "title": "3 of 4 tests passed",
  },
  "status": "completed",
}
`;

exports[`checks/jest should handle skipped tests 1`] = `
Object {
  "completed_at": StringMatching /\\^\\\\d\\{4\\}/,
  "conclusion": "success",
  "head_sha": "1234567890",
  "name": "jest",
  "output": Object {
    "annotations": Array [
      Object {
        "annotation_level": "notice",
        "end_line": 1,
        "message": "skipped test 'is possible to distinguish between duplicates and all other errors'",
        "path": "src/tests/integration/insert-ignore.test.ts",
        "raw_details": "{
  \\"ancestorTitles\\": [
    \\"Tests INSERT IGNORE warnings\\"
  ],
  \\"failureMessages\\": [],
  \\"fullName\\": \\"Tests INSERT IGNORE warnings is possible to distinguish between duplicates and all other errors\\",
  \\"location\\": null,
  \\"status\\": \\"pending\\",
  \\"title\\": \\"is possible to distinguish between duplicates and all other errors\\",
  \\"file\\": \\"/Users/tlb/git/connectedcars/data/src/tests/integration/insert-ignore.test.ts\\"
}",
        "start_line": 1,
        "title": "Tests INSERT IGNORE warnings - is possible to distinguish between duplicates and all other errors",
      },
    ],
    "summary": "8 of 9 tests passed",
    "title": "8 of 9 tests passed",
  },
  "status": "completed",
}
`;
